components/plugin_marketplace/marketplace_item/marketplace_item_plugin/marketplace_item_plugin.tsx

```diff
@@ -301,39 +365,54 @@ export default class MarketplaceItemPlugin extends React.PureComponent <Marketpl
                     </button>
                 </Link>
             );
-        }
-
-        let actionButton: JSX.Element;
-        if (this.props.error) {
-            actionButton = (
-                <FormattedMessage
-                    id='marketplace_modal.list.try_again'
-                    defaultMessage='Try Again'
-                />
-            );
         } else {
+            let actionLabel: JSX.Element;
+            if (this.props.error) {
+                actionLabel = (
+                    <FormattedMessage
+                        id='marketplace_modal.list.try_again'
+                        defaultMessage='Try Again'
+                    />
+                );
+            } else {
+                actionLabel = (
+                    <FormattedMessage
+                        id='marketplace_modal.list.install'
+                        defaultMessage='Install'
+                    />
+                );
+            }
+
+            let disableInstallButton = this.props.installing;
+            const limit = this.props.limits?.integrations?.enabled;
+            const usage = this.props.integrationsUsage?.enabled;
+            if (limit && usage && usage >= limit) {
+                disableInstallButton = true;
+            }
+
             actionButton = (
-                <FormattedMessage
-                    id='marketplace_modal.list.install'
-                    defaultMessage='Install'
-                />
+                <button
+                    onClick={this.onInstall}
+                    className='btn btn-primary'
+                    disabled={disableInstallButton}
+                >
+                    <LoadingWrapper
+                        loading={this.props.installing}
+                        text={localizeMessage('marketplace_modal.installing', 'Installing...')}
```

Comment:
This is pre-existing so this comment is optional, but could we update this to be a `<FormattedMessage/>`? `localizeMessage` does some things wrong and is deprecated.